database/redis/selfstate_test.go

```diff
@@ -65,7 +66,7 @@ func TestSelfCheckWithWritesNotInChecker(t *testing.T) {
 
 func testSelfCheckWithWritesInDBSource(t *testing.T, dbSource DBSource) {
 	logger, _ := logging.GetLogger("dataBase")
-	dataBase := NewDatabase(logger, config, dbSource)
+	dataBase := NewTestDatabase(logger)
```

Comment:
Probably you'd want to add like you add above for `Checker`:
```
	dataBase.source = dbSource 
```

Reply:
fixed